# G2: Example 2, multi-channel experiments

**This example** describes a set of solutions used in experiments on multiple ion channels.
They were prepared and pH adjusted at a room temperature of $20.5,^\circ\text{C}$, and patch clamp experiments were performed at the same temperature.

**Summary**: In this example, the compounds are mostly very pure but total volume is set with a 2% precision, which becomes the main source of error.
We briefly discuss the accuracy of scales and the people using them (in this case the author).

## The bath solution

| Compound | $C_t$    | $M$          | $p$           | $C_s$ | Product |
|----------|----------|--------------|---------------|-------|---------|
| NaCl     | 140 mM   |  58.44 g/mol | $\geq 99.5\%$ |       | [S7653](https://www.sigmaaldrich.com/GB/en/product/sial/s7653) |
| KCl      |   5.4 mM |  74.55 g/mol | $\geq 99\%$   |       | [P9333](https://www.sigmaaldrich.com/GB/en/product/sial/p9333) |
| MgCl2    |   1 mM   |              | $\geq 99\%$   | 1 M * | [105832](https://www.sigmaaldrich.com/GB/en/product/mm/105832) |
| CaCl2    |   1 mM   |              | $\geq 99\%$   | 1 M * | [102382](https://www.sigmaaldrich.com/GB/en/product/mm/102382) |
| HEPES    |  15 mM   | 238.3  g/mol | $\geq 99.5\%$ |       | [H3375](https://www.sigmaaldrich.com/GB/en/product/sigma/h3375) |
| Glucose  |   6 mM   | 180.16 g/mol | $\geq 99.5\%$ |       | [G8270](https://www.sigmaaldrich.com/GB/en/product/sigma/g8270) |

To add $\text{MgCl}_2$ and $\text{CaCl}_2$ a 1M solution was prepared from powder, and then added by volume.

### pH adjustment and volume change

The solution was mixed in a 1L volume, and adjusted to pH 7.4 using 7.7 mM NaOH.

Specifically, 0.6 mL 10 M NaOH, and 1.7 mL 1 M NaOH.
\begin{align}
  0.6\,\text{mL} \cdot 10\,\frac{\text{mol}}{\text{L}} 
+ 1.7\,\text{mL} \cdot 1\,\frac{\text{mol}}{\text{L}}
= 7.7\,\text{mmol} \\
\end{align}

This happened **after** setting the final volume, which therefore increased by 2.3 mL.

### Simple ions

The salts ($\text{NaCl}$, $\text{KCl}$, $\text{MgCl}_2$, and $\text{CaCl}_2$) will fully dissolve, and no substances were added that will bind the dissociated ions.
Therefore, we have a total of 147.7 mM $\text{Na}^+$, 5.4 mM $\text{K}^+$, 1 mM $\text{Ca}^{2+}$, 1 mM $\text{Mg}^{2+}$, and $140 + 5.4 + 2 \cdot 2 = 149.4\,\text{mM}$ $\text{Cl}^-$.

### pH buffer

HEPES' pKa is 7.55 at $20\,^\circ\text{C}$ and shifts by -0.014 per $^\circ\text{C}$, for a value of 7.543 at $20.5\,^\circ\text{C}$:
\begin{align}
[\text{HEPES}^-] = \frac{[\text{HEPES}]_\text{added}}{1 + 10^{\text{pK}_a - \text{pH}}} \approx 6.28\,\text{mM} (41.8\%)
\end{align}

### Free ion concentrations

| Solute           | mM added | mM free | Total charge | Total solutes |
|------------------|----------|---------|--------------|---------------|
| $\text{Na}^+$    | 147.7    | 147.7   | +147.7       | 147.7         |
| $\text{K}^+$     |   5.4    |   5.4   | +153.1       | 153.1         |
| $\text{Cl}^-$    | 149.4    | 149.4   | +  3.7       | 302.5         |
| $\text{Ca}^{2+}$ |   1      |   1     | +  5.7       | 303.5         |
| $\text{Mg}^{2+}$ |   1      |   1     | +  7.7       | 304.5         |
| $\text{HEPES}$   |  15      |   8.7   |              | 313.2         |
| $\text{HEPES}^-$ |          |   6.3   | +  1.4       | 319.5         |
| Glucose          |   6      |   6     |              | 325.5         |

Here the final column shows the solution's osmolarity.

### Charge balance

The final charge balance is off by 1.4 mM.

Since NaCl, KCl, MgCl2 and CaCl2 are all electrically neutral and do not react with water, they cannot contribute to this error, which instead must be due to:

- the NaOH measurement or stock concentrations (mL added, concentration of added solution, total volume)
- the HEPES concentration (weight, pH, total volume)


## Pipette solution

| Compound    | $C_t$    | $M$          | $p$           | $C_s$ | $m_a$   | Product |
|-------------|----------|--------------|---------------|-------|---------|---------|
| KCl         |  10 mM   |  74.55 g/mol | $\geq 99\%$   |       |   74 mg | [P9333](https://www.sigmaaldrich.com/GB/en/product/sial/p9333) |
| K-Gluconate | 125 mM   | 234.25 g/mol | $\geq 99\%$   |       | 2929 mg | [860379](https://www.sigmaaldrich.com/GB/en/product/aldrich/860379) |
| MgCl2       |   2 mM   |              | $\geq 99\%$   | 1 M   |         | [105832](https://www.sigmaaldrich.com/GB/en/product/mm/105832) |
| CaCl2       |   0.6 mM |              | $\geq 99\%$   | 1 M   |         | [102382](https://www.sigmaaldrich.com/GB/en/product/mm/102382) |
| EGTA        |   5 mM   | 380.35 g/mol | $\geq 97\%$   |       |  119 mg | [E4378](https://www.sigmaaldrich.com/GB/en/product/sial/e4378) |
| Na2ATP      |   4 mM   | 551.1 g/mol  | $\geq 99\%$ * |       | 1027 mg | [A26209](https://www.sigmaaldrich.com/GB/en/product/aldrich/a26209) |
| HEPES       |   5 mM   | 238.3 g/mol  | $\geq 99.5\%$ |       | 220 mg | [H3375](https://www.sigmaaldrich.com/GB/en/product/sigma/h3375) |
| Sucrose     |  30 mM   | 342.30 g/mol | $\geq 99.5\%$ |       | 191 mg | [S9378](https://www.sigmaaldrich.com/GB/en/product/sigma/s9378) | 

Although $\text{Na}_2\text{ATP}$ has a high purity, it also has significant water content, indicated by an asterisk in the table.
Like the other compounds, we will initially treat it as pure, and deal with the water content when delving into uncertainties.

### pH adjustment and volume change

The solution was mixed in a 100mL volume, and adjusted to pH 7.2 using 19.2 mM KOH.
Specifically, 0.18 mL 10 M KOH, and 0.12 mL 1 M KOH.

This happened **after** setting the final volume, which therefore increased by 0.3 mL.

### Simple ions

For potassium and chloride we can immediately write down $[\text{K}^+] = 10 + 125 + 19.2 = 154.2\,\text{mM}$ and $[\text{Cl}^-] = 10 + 2 \cdot 2 + 2 \cdot 0.6 = 15.2\,\text{mM}$.
For sodium, we use $[\text{Na}^+]=8\,\text{mM}$

### pH buffer

For HEPES, we use pKa=7.543 and pH=7.2 for
\begin{align}
[\text{HEPES}^-] = \frac{[\text{HEPES}]_\text{added}}{1 + 10^{\text{pK}_a - \text{pH}}} \approx 1.56\,\text{mM} (31.2\%)
\end{align}

### Gluconate

For Gluconate, we use [pKa=3.86](https://en.wikipedia.org/wiki/Gluconic_acid) and pH=7.2 for
\begin{align}
[\text{Gluconate}^-] = \frac{[\text{Gluconate}]_\text{added}}{1 + 10^{\text{pK}_a - \text{pH}}} \approx 124.94 \approx 125\text{mM}
\end{align}

### Ca and EGTA

For Ca and EGTA we use the simplifying assumptions to find 0 mM $\text{Ca}^{2+}$, 0.6 mM $\text{CaEGTA}^{2-}$, and 4.4 mM $\text{H}_2\text{EGTA}^{2-}$.

### Mg and ATP

For Mg and ATP we first look up whether $\text{Na}_2\text{ATP}$ dissolves (it does).
Here we have more ATP than $\text{Mg}^{2+}$, so will assume 2 mM $\text{MgATP}^{2-}$, 2 mM free ATP, and no free $\text{Mg}^{2+}$.

Using
\begin{align}
[\text{ATP}^{4-}] = \frac{[\text{ATP}]_\text{free}}{1 + 10^{\text{pK}_a - \text{pH}}}
&&\quad&&
[\text{HATP}^{3-}] = \frac{[\text{ATP}]_\text{free}}{1 + 10^{\text{pH} - \text{pK}_a}}
\end{align}
with $\text{pH}=7.2$ and $\text{pKa}=6.5$ ([Stockbridge & Wolfenden](https://doi.org/10.1074/jbc.M109.017806)) we find $[\text{ATP}^{4-}]=1.67\,\text{mM}$ (83.5%) and $[\text{HATP}^{3-}]=0.33\,\text{mM}$ (16.5%).

### Free ion concentrations

| Solute                       | mM added | mM free | Total charge | Total abs(charge)| Total solutes |
|------------------------------|----------|---------|--------------|------------------|---------------|
| $\text{Na}^+$                |   8      |   8     | +   8        |   8              |   8           |
| $\text{K}^+$                 | 154.2    | 154.2   | + 162.2      | 162.2            | 162.2         |
| $\text{Cl}^-$                |  15.2    |  15.2   | + 147        | 177.4            | 177.4         |
| $\text{Ca}^{2+}$             |   0.6    |         |              |                  |               |
| $\text{CaEGTA}^{2-}$         |          |   0.6   | + 145.8      | 178.6            | 178           |
| $\text{H}_2\text{EGTA}^{2-}$ |          |   4.4   | + 137        | 187.4            | 182.4         |
| $\text{H}_4\text{EGTA}$      |   5      |         |              |                  |               |
| $\text{Mg}^{2+}$             |   2      |         |              |                  |               |
| $\text{MgATP}^{2-}$          |          |   2     | + 133        | 191.4            | 184.4         |
| $\text{HATP}^{3-}$           |          |   0.33  | + 132.01     | 192.39           | 184.73        |
| $\text{ATP}^{4-}$            |          |   1.67  | + 125.33     | 199.07           | 186.4         |
| $\text{HEPES}$               |   5      |   3.4   | +            |                  | 189.8         |
| $\text{HEPES}^-$             |          |   1.6   | + 123.73     | 200.67           | 191.4         |
| $\text{Gluconate}$           | 125      |         |              |                  |               |
| $\text{Gluconate}^-$         |          | 125     | - 1.27       | 325.67           | 316.4         |


## Combined table

We can now table the free ion concentrations from both solutions:

| Ion                          | mM int | mM ext | Mobility         |
|------------------------------|--------|--------|------------------|
| $\text{Na}^+$                |   8    | 147.7  | In LJPCalc       |
| $\text{K}^+$                 | 154.2  |   5.4  | In LJPCalc       |
| $\text{Cl}^-$                |  15.2  | 149.4  | In LJPCalc       |
| $\text{Ca}^{2+}$             |        |   1    | In LJPCalc       |
| $\text{CaEGTA}^{2-}$         |   0.6  |        | Approx EGTA(2-)  |
| $\text{H}_2\text{EGTA}^{2-}$ |   4.4  |        | In LJPCalc       |
| $\text{Mg}^{2+}$             |        |   1    | In LJPCalc       |
| $\text{MgATP}^{2-}$          |   2    |        | Approx ATP(2-)   |
| $\text{HATP}^{3-}$           |   0.33 |        | In LJPCalc       |
| $\text{ATP}^{4-}$            |   1.67 |        | In LJPCalc       |
| $\text{HEPES}^-$             |   1.6  |   6.3  | In LJPCalc       |
| $\text{Gluconate}^-$         | 125    |        | In LJPCalc       |


## Liquid junction potential

To calculate the LJP, we use [LJPCalc](https://swharden.com/LJPcalc/) with the following input

```
ion: Na, +1, 50.11, 8, 147.7
ion: K, +1, 73.5, 154.2, 5.4
ion: Cl, -1, 76.31, 15.2, 149.4
ion: Ca, +2, 59.5, 0, 1
ion: EGTA(2-), -2, 35.28, 5, 0
ion: Mg, +2, 53.06, 0, 1
ion: ATP (Adenosine 5'-Triphosphate), -2, 22.05, 2, 0
ion: ATP (Adenosine 5'-Triphosphate), -3, 33.075, 0.33, 0
ion: ATP (Adenosine 5'-Triphosphate), -4, 44.1, 1.67, 0
ion: HEPES, -1, 22.05, 1.6, 6.3
ion: Gluconate, -1, 24.255, 125, 0
temperatureC: 20.5
```

This yields an LJP of +15.8 mV.

## Reversal potentials

The reversal potentials for sodium and potassium are

In [1]:
import math

T = 273.15 + 20.5
R = 8314  # J/kmol/K
F = 96485 # C/mol
print(f'ENa {R * T / F * math.log(147.7 / 8): 5.1f}')
print(f'EK  {R * T / F * math.log(5.4 / 154.2): 5.1f}')

ENa  73.8
EK  -84.8


## Osmolarity

The osmolarity in the external solution is 325.5, compared to 316.4 internally.

## Uncertainty in the target concentrations

Finally, we look at uncertainty in the target concentrations that arise when measuring out ingredients.

### Adding by weight

To estimate the uncertainty in concentration for substances added by weight, we need to assign values to the terms in
\begin{align}
\frac{\Delta}{C} &= \sqrt{\left(\frac{\Delta m_a}{m_a}\right)^2 + \left(\frac{\Delta V}{V}\right)^2 + \left(\frac{\Delta q}{q}\right)^2}
\end{align}

#### $m_a$ and $\Delta m_a$

In this experiment, the added weights $m_a$ were recorded.
The table below lists the intended weights, $m$, in addition to actually added weight (read directly from the scales), for the external solution:

| Compound | $C_t$    | $M$          | $p$           | $m$       | $m_a$   |
|----------|----------|--------------|---------------|-----------|---------|
| NaCl     | 140 mM   |  58.44 g/mol | $\geq 99.5\%$ | 8181.6 mg | 8178 mg |
| KCl      |   5.4 mM |  74.55 g/mol | $\geq 99\%$   |  402.6 mg |  402 mg |
| HEPES    |  15 mM   | 238.3  g/mol | $\geq 99.5\%$ | 3574.5 mg | 3570 mg |

The internal solution was made more carefully:

| Compound    | $C_t$    | $M$          | $p$           | $m$       | $m_a$     |
|-------------|----------|--------------|---------------|-----------|-----------|
| KCl         |  10 mM   |  74.55 g/mol | $\geq 99\%$   |   74.6 mg |   74.4 mg |
| K-Gluconate | 125 mM   | 234.25 g/mol | $\geq 99\%$   | 2928.1 mg | 2929.0 mg |
| EGTA        |   5 mM   | 380.35 g/mol | $\geq 97\%$   |  190.2 mg |  190.7 mg |
| Na2ATP      |   4 mM   | 551.1 g/mol  | $\geq 99\%$ * |  220.4 mg |  220.6 mg |
| HEPES       |   5 mM   | 238.3 g/mol  | $\geq 99.5\%$ |  119.2 mg |  119.5 mg |

For $\Delta m_a$ we can look at specifications of "analytical balances", for example [this one](https://www.mt.com/us/en/home/phased_out_products/Laboratory_Weighing_Solutions/Analytical/Excellence/XS_Analytical_Balance/XS64.html) used in Example 2.
It shows several measures, including readability (how many digits are on the display, regardless of accuracy), repeatability (the standard deviation obtained if you keep measuring the same thing), and linearity (if you measure objects over the full allowable range, what is the maximum deviation from the true weight).
To find an upper bound, we could use the largest listed number of $\pm0.2\,\text{mg}$.
However, this precision is only achieved if the scales are used with extreme care, so that something larger, for example $\pm1\,\text{mg}$ may be more realistic.

In addition to precision of the _scales_, we can factor precision of the experimenter into $\Delta m_a$.
Above, the experimenter accepted errors of 4, 0.6, and 4.5 mg in the external solution, and, on a different day, errors of 0.2, 0.9, 0.5, 0.2, and 0.3 mg.
We can use these numbers to set an approximate $\Delta m_a$ in cases where $m_a$ is unknown, using $\Delta m_a = 1\,\text{mg}$ for a precisely made solution (so that the error of the scales dominates), and $\Delta m_a = 5\,\text{mg}$ in a less precisely made solution.

#### $\Delta V / V$

In this experiment, the external volume was set using a 1L chemical beaker, with an accuracy of 20mL, giving
\begin{align}
\frac{\Delta V}{V} = \frac{20}{1000} = 0.02
\end{align}

#### Purity

Because purity was not taken into account, we will incur both a bias and a relative error due to $p$.

We will factor in purity using 
\begin{align}
C = \frac{m_a \cdot q}{M \cdot V}
& \quad &
q = \frac{1 + p}{2}
& \quad &
\Delta q = \frac{1 - p}{2}
& \quad &
\frac{\Delta q}{q} = \frac{1 - p}{1 + p}
\end{align}

### External solution

Filling in for NaCl, KCl and HEPES, we find:

In [2]:
# Ct, M, purity, ma
ions = {
    'NaCl': (140, 58.44, 99.5, 8178),
    'KCl': (5.4, 74.55, 99, 402),
    'HEPES': (15, 238.3, 99.5, 3570),
}

# Volume and relative error in final volume
V = 1
dVV = 0.02

# Absolute error in weight
dm = 1

for k, (Ct, M, p, ma) in ions.items():
    p = p / 100
    q = (1 + p) / 2
    dq = (1 - p) / 2
    C = ma * q / (M * V)
    print(f'{k} (Ct={Ct} mM)')
    dcc = math.sqrt(dVV**2 + (dm / ma)**2 + (dq / q)**2)
    dc = dcc * C
    print(f'  |dV/V|: {abs(dVV):1.6g}')
    print(f'  |dm/m|: {abs(dm / ma):1.6g}')
    print(f'  |dq/q|: {abs(dq / q):1.6g}')
    print(f'   dC/C : {dcc:1.6f}')
    print(f'   dC   : {dc:1.6f} mM')
    print(f'    C   : {C:6.2f} [{C - dc:6.2f}, {C + dc:5.1f}]')


NaCl (Ct=140 mM)
  |dV/V|: 0.02
  |dm/m|: 0.000122279
  |dq/q|: 0.00250627
   dC/C : 0.020157
   dC   : 2.813658 mM
    C   : 139.59 [136.77, 142.4]
KCl (Ct=5.4 mM)
  |dV/V|: 0.02
  |dm/m|: 0.00248756
  |dq/q|: 0.00502513
   dC/C : 0.020771
   dC   : 0.111445 mM
    C   :   5.37 [  5.25,   5.5]
HEPES (Ct=15 mM)
  |dV/V|: 0.02
  |dm/m|: 0.000280112
  |dq/q|: 0.00250627
   dC/C : 0.020158
   dC   : 0.301240 mM
    C   :  14.94 [ 14.64,  15.2]


This shows that, in this example:

1. The uncertainty in total volume dominates, and the relative errors in NaCl, KCl, and HEPES concentration are all 2 to 2.1%.
2. Because of the high purities used, the bias due to not correcting for purity is low.
3. The uncertainty in added weight is is negligible, even for KCl which had the smallest added weight

The volume increase from pH adjustment was 1.15 mL, and not included in the analysis above.
But as this was a 0.1% increase, it will be dwarfed by the 2% error and so is safely ignored.

### Adding by volume

The substances added by volume were added with a precision pipette, and so here as well the 2% volume error will dominate.

### Internal solution

To repeat this for the internal solution, we need to deal with the unspecified water content in $\text{Na}_2\text{ATP}$.
Although we do not have the lot number for this data, we can look at the 5 last published certificates of analysis (using the product link above).
Here, we see that

1. The upper limit is 12% water
2. The last five water lots have water content 7.1%, 6.4%, 6.4%, 9.0%, 4.7%.

So it looks like values are spread out over the interval, not clustered near the upper limit.
An easy way to incorporate the water content then, is to ignore the higher purity (99%), and fill in $q=0.88$ instead.

We then find:

In [3]:
# Ct, M, purity, ma
ions = {
    'KCl': (10, 74.55, 99, 74.4),
    'K-Gluconate': (125, 234.25, 99, 2929),
    'EGTA': (5, 380.35, 96, 190.7),  # Combined purity and lod
    'Na2ATP': (4, 551.1, 88, 220.6),
    'HEPES': (5, 238.3, 99.5, 119.5),
}

# Volume and relative error in final volume
V = 1
dVV = 0.02

# Absolute error in weight
dm = 1

for k, (Ct, M, p, ma) in ions.items():
    p = p / 100
    q = (1 + p) / 2
    dq = (1 - p) / 2
    C = ma * q / (M * V) / 0.1
    print(f'{k} (Ct={Ct} mM)')
    dcc = math.sqrt(dVV**2 + (dm / ma)**2 + (dq / q)**2)
    dc = dcc * C
    print(f'  |dV/V|: {abs(dVV):1.6g}')
    print(f'  |dm/m|: {abs(dm / ma):1.6g}')
    print(f'  |dq/q|: {abs(dq / q):1.6g}')
    print(f'   dC/C : {dcc:1.6f}')
    print(f'   dC   : {dc:1.6f} mM')
    print(f'    C   : {C:6.2f} [{C - dc:6.2f}, {C + dc:5.1f}]')


KCl (Ct=10 mM)
  |dV/V|: 0.02
  |dm/m|: 0.0134409
  |dq/q|: 0.00502513
   dC/C : 0.024615
   dC   : 0.244429 mM
    C   :   9.93 [  9.69,  10.2]
K-Gluconate (Ct=125 mM)
  |dV/V|: 0.02
  |dm/m|: 0.000341413
  |dq/q|: 0.00502513
   dC/C : 0.020624
   dC   : 2.565934 mM
    C   : 124.41 [121.85, 127.0]
EGTA (Ct=5 mM)
  |dV/V|: 0.02
  |dm/m|: 0.00524384
  |dq/q|: 0.0204082
   dC/C : 0.029052
   dC   : 0.142745 mM
    C   :   4.91 [  4.77,   5.1]
Na2ATP (Ct=4 mM)
  |dV/V|: 0.02
  |dm/m|: 0.00453309
  |dq/q|: 0.0638298
   dC/C : 0.067043
   dC   : 0.252265 mM
    C   :   3.76 [  3.51,   4.0]
HEPES (Ct=5 mM)
  |dV/V|: 0.02
  |dm/m|: 0.0083682
  |dq/q|: 0.00250627
   dC/C : 0.021824
   dC   : 0.109169 mM
    C   :   5.00 [  4.89,   5.1]


## Effect of uncertainties

A combined table of ranges for each free ion is given below.

**Because most of the errors are due to the total volume adjustment, they are strongly correlated: if one of the internal free ions is at the low side of the bracket, then all internal ions must be.**
And the same holds for the external ions, although there is no (or very little) correlation between internal and external.
The main exceptions are internal sodium and ATP, which have a strong independent variation.

| Ion                          | mM int         | mM ext         |
|------------------------------|----------------|----------------|
| $\text{Na}^+$                | [7.0, 8.0]     | [144.5, 150.1] |
| $\text{K}^+$                 | [150.7, 156.4] | [5.25, 5.5]    |
| $\text{Cl}^-$                | [14.8, 15.5]   | [145.9, 152.0] |
| $\text{Ca}^{2+}$             |                | [0.98, 1.02]   |
| $\text{CaEGTA}^{2-}$         | [0.59, 0.61]   |                |
| $\text{H}_2\text{EGTA}^{2-}$ | [4.16, 4.51]   |                |
| $\text{Mg}^{2+}$             |                | [0.98, 1.02]   |
| $\text{MgATP}^{2-}$          | [1.96, 2.04]   |                |
| $\text{HATP}^{3-}$           | [0.24, 0.34]   |                |
| $\text{ATP}^{4-}$            | [1.23, 1.70]   |                |
| $\text{HEPES}^-$             | [1.53, 1.59]   | [6.13, 6.36]   |
| $\text{Gluconate}^-$         | [121.9, 127.0] |                |

Details:

- The sodium and potassium brackets include the amount added by titration (with no extra error)
- The $\text{CaCl}_2$ and $\text{MgCl}_2$ brackets are based on a 2% error
- The free chloride concentration was summed from the various brackets containing it
- The deprotonated HEPES was recalculated
- The CaEGTA bracket was set to the Ca bracket
- The EGTA bracket was based on the Ca and EGTA brackets
- The MgATP bracket was set to the Mg bracket
- A free ATP bracket was calculated from the ATP and Mg brackets, and split into 16.5% and 83.5%


### Reversal potentials

Using the error bounds on concentrations, we can explore the reversal potentials:

In [4]:
ENa = R * T / F * math.log(148 / 8)
Elo = R * T / F * math.log(144.5 / 8)
Ehi = R * T / F * math.log(150.1 / 7)

print(f'ENa: {ENa:.4} [{Elo:.4}, {Ehi:.4}]')
print(f' dE: {max(Ehi - ENa, ENa - Elo):.2}')

ENa: 73.83 [73.22, 77.57]
 dE: 3.7


In [5]:
EK  = R * T / F * math.log(5.4 / 154.2)
Elo = R * T / F * math.log(5.25 / 156.4)
Ehi = R * T / F * math.log(5.5 / 150.7)

print(f'EK: {EK:.4} [{Elo:.4}, {Ehi:.4}]')
print(f'dE: {max(Ehi - EK, EK - Elo):.2}')

EK: -84.81 [-85.88, -83.77]
dE: 1.1


Or alternatively a $\pm3.7\,\text{mV}$ error in $\text{E}_\text{Na}$, and a $\pm1.1\,\text{mV}$ error in $\text{E}_\text{K}$.

### Liquid junction potential

Assuming only correlated errors, we can explore both edges of the brackets with the following LJP input:

```
ion: Na, +1, 50.11, 8, 144.5
ion: K, +1, 73.5, 156.4, 5.25
ion: Cl, -1, 76.31, 15.5, 152
ion: Ca, +2, 59.5, 0, 0.98
ion: EGTA(2-), -2, 35.28, 5.12, 0
ion: Mg, +2, 53.06, 0, 0.98
ion: ATP (Adenosine 5'-Triphosphate), -2, 22.05, 2.04, 0
ion: ATP (Adenosine 5'-Triphosphate), -3, 33.075, 0.34, 0
ion: ATP (Adenosine 5'-Triphosphate), -4, 44.1, 1.70, 0
ion: HEPES, -1, 22.05, 1.59, 6.13
ion: Gluconate, -1, 24.255, 127, 0
temperatureC: 20.5
```
Which yields +15.9mV

and
```
ion: Na, +1, 50.11, 7, 150.1
ion: K, +1, 73.5, 150.7, 5.5
ion: Cl, -1, 76.31, 14.8, 153
ion: Ca, +2, 59.5, 0, 1.02
ion: EGTA(2-), -2, 35.28, 4.75, 0
ion: Mg, +2, 53.06, 0, 1.02
ion: ATP (Adenosine 5'-Triphosphate), -2, 22.05, 1.96, 0
ion: ATP (Adenosine 5'-Triphosphate), -3, 33.075, 0.24, 0
ion: ATP (Adenosine 5'-Triphosphate), -4, 44.1, 1.23, 0
ion: HEPES, -1, 22.05, 1.53, 6.36
ion: Gluconate, -1, 24.255, 121.9, 0
temperatureC: 20.5
```
Which yields +15.5mV

For a combined LJP estimate of +15.8 mV [15.5, 15.9].